### PR TITLE
CVSB-3005

### PR DESCRIPTION
**Pull request by vladddragomir**

### CODE REVIEW

File: src/pages/testing/test-creation/eu-vehicle-category/eu-vehicle-category.ts [hunk 1]
```diff
@@ -11,21 +11,25 @@ import { VisitService } from "../../../../providers/visit/visit.service";
 })
 export class CategoryReadingPage implements OnInit {
   vehicle: VehicleModel;
+  errorIncomplete: boolean;
```
Comment (agoncearenco) [inline src/pages/testing/test-creation/eu-vehicle-category/eu-vehicle-category.ts]: Why do you need the errorIncomplete flag in the visit service and also in each individual page? Couldn't you use this flag everywhere in the app?

File: src/pages/testing/test-creation/test-type-details-input/test-type-details-input.ts [hunk 2]
```diff
@@ -16,21 +16,23 @@ export class TestTypeDetailsInputPage implements OnInit {
...
-  ngOnInit() {
     this.vehicleCategory = this.navParams.get('vehicleCategory');
     this.sectionName = this.navParams.get('sectionName');
     this.input = this.navParams.get('input');
     this.fromTestReview = this.navParams.get('fromTestReview');
     this.inputValue = this.navParams.get('existentValue');
+    this.errorIncomplete = this.navParams.get('errorIncomplete');
+  }
+
+  ngOnInit() {
```
Comment (agoncearenco) [inline src/pages/testing/test-creation/test-type-details-input/test-type-details-input.ts]: Why did you need to move all these initializations in the constructor?

File: src/pages/testing/test-creation/eu-vehicle-category/eu-vehicle-category.ts (hunk 1 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/eu-vehicle-category/eu-vehicle-category.ts]: i need it separately because i need different states on each component

File: src/pages/testing/test-creation/test-type-details-input/test-type-details-input.ts (hunk 2 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-type-details-input/test-type-details-input.ts]: because that's the best place to get data from navparams. ngOnInit could sometimes be too late and therefore make the template crash.

Comment (agoncearenco) [reply src/pages/testing/test-creation/test-type-details-input/test-type-details-input.ts]: So you're saying it wasn't working correctly before?

File: src/pages/testing/test-creation/eu-vehicle-category/eu-vehicle-category.ts (hunk 1 above)
Comment (agoncearenco) [reply src/pages/testing/test-creation/eu-vehicle-category/eu-vehicle-category.ts]: You sure you need different states? This might lead to difficulties with managing the validation

Comment (vladddragomir) [reply src/pages/testing/test-creation/eu-vehicle-category/eu-vehicle-category.ts]: yes, i'm sure and it also should not cause any difficulties if the requirements don't change. i can't use a service level variable because while on test-create page the error should be global, when entering the secondary pages that error should be able to inactivate itself and at the same time when returning to test-create page should still be activated if not all mandatory fields are captured. so, yes, a service level variable won't work!

File: src/pages/testing/test-creation/test-type-details-input/test-type-details-input.ts (hunk 2 above)
Comment (vladddragomir) [reply src/pages/testing/test-creation/test-type-details-input/test-type-details-input.ts]: yes, it was. but it's more safe to keep them in the constructor because there is a small chance that sometime the app breaks
